### DISCUSSION TITLE
Spout library not found. Try setting SPOUT_LIBRARY or disable Spout support

### DISCUSSION THREAD

**Discussion by BikramAjay** [Stellarium/stellarium#1802] [General]
"E:\...\stellarium-0.21.0\CMakeLists.txt:343: error: Spout library not found.  Try setting SPOUT_LIBRARY or disable Spout support."

I am trying to compile Stellarium source code in Qt and getting the abv error. 

I have downloaded Spout from
https://spout.zeal.co/
and installed at 
C:\Users\xxxx\SPOUT-2007

as instructed in its documentation.

But the abv error message doesn't go away.

Has anyone faced this problem and found a way to fix it?
Any help is greatly appreciated.

**Comment by gzotti** [comment]
Why did you install another copy of Spout in addition to the included copy in util/spout2?

**Reply by BikramAjay** [reply]
Until I read your reply I didn't know it's already included in util. So after seeing the error I did a Google search and installed Spout as directed in its documentation. 

But I don't see any spout2 in the util directory. Please see attached screenshots.

![Screenshot 2021-07-18 203041](https://user-images.githubusercontent.com/86772525/126087206-18e5c3b5-6bd9-4fde-add7-3b11869fa361.jpg)
![Screenshot 2021-07-18 203121](https://user-images.githubusercontent.com/86772525/126087222-f429523c-25ae-4a16-86ec-65e3fdcf258d.jpg)

**Reply by gzotti** [reply]
The project config probably has not included it because something else was wrong. (See first line of your OP)

Occasionally you must help cmake. Find CMakeCache.txt in your build directory and fill in:

```
//Path to a library.
SPOUT_LIBRARY:FILEPATH=D:/StelDev/GIT/stellarium/util/spout2/x64/SpoutLibrary.lib

//The path to the Spout .dll file
SPOUT_LIBRARY_DLL:FILEPATH=D:/StelDev/GIT/stellarium/util/spout2/x64/SpoutLibrary.dll

//The path to the Spout licence file
SPOUT_LIBRARY_LICENCE:FILEPATH=D:/StelDev/GIT/stellarium/util/spout2/licence.txt
```

Then re-run cmake.

**Reply by BikramAjay** [reply]
Thanks but I am working on the Windows platform.

**Comment by alex-w** [comment]
Did you tried build Stellarium from tar.gz package or from git master?

**Reply by BikramAjay** [reply]
from tar.gz package

**Reply by alex-w** [reply]
OK, spout library is not included in tar.gz package because it uses almost on linux/unix systems. Of course in this case spout library should be downloaded and available through PATH variable.

**Reply by BikramAjay** [reply]
Thanks for the reply. For the compilation on Windows one should download (in case he doesn't want to use Git) from the following link.  It has spout lib included in it. 

https://github.com/Stellarium/stellarium/archive/master.zip


Screenshot from here
https://github.com/Stellarium/stellarium/blob/master/BUILDING.md
![Screenshot 2021-07-20 231344](https://user-images.githubusercontent.com/86772525/126424965-739bf6df-0efe-4621-9c63-ac297d3ef9c3.jpg)

**Reply by BikramAjay** [reply]
I am getting this new error message though. My Qt version is 6.2.o and it seems the CMake file is specifically looking for Qt version 5.x.x. 

"C:\Devel\stellarium\CMakeLists.txt:464: error: By not providing "FindQt5Core.cmake" in CMAKE_MODULE_PATH this project has asked CMake to find a package configuration file provided by "Qt5Core", but CMake did not find one. Could not find a package configuration file provided by "Qt5Core" with any of the following names: Qt5CoreConfig.cmake qt5core-config.cmake Add the installation prefix of "Qt5Core" to CMAKE_PREFIX_PATH or set "Qt5Core_DIR" to a directory containing one of the above files.  If "Qt5Core" provides a separate development package or SDK, be sure it has been installed."

**Reply by alex-w** [reply]
Stellarium want Qt 5, not Qt 6!

**Reply by gzotti** [reply]
Stellarium is not ready for Qt6. This will be a major step to take (hopefully) during the next year. If you are very familiar with Qt, we can also accept help!
